### PR TITLE
CORDA-540: Add a property to make sending stacktraces optional in AMQP mode

### PR DESCRIPTION
**Pull request by vkolomeyko**

### CODE REVIEW

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/custom/ThrowableSerializer.kt [hunk 1]
```diff
@@ -33,7 +35,18 @@ class ThrowableSerializer(factory: SerializerFactory) : CustomSerializer.Proxy<T
...
         }
-        return ThrowableProxy(obj.javaClass.name, message, obj.stackTrace, obj.cause, obj.suppressed, extraProperties)
+        return if(shouldIncludeInternalInfo())
+                    ThrowableProxy(obj.javaClass.name, message, obj.stackTrace, obj.cause, obj.suppressed, extraProperties)
+                else
+                    ThrowableProxy(obj.javaClass.name, message, emptyArray(), obj.cause, obj.suppressed, extraProperties)
+    }
+
+    private fun shouldIncludeInternalInfo(): Boolean {
+
```
Comment (rick-r3) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/custom/ThrowableSerializer.kt]: remove blank line

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/custom/ThrowableSerializer.kt [hunk 2]
```diff
@@ -33,7 +35,18 @@ class ThrowableSerializer(factory: SerializerFactory) : CustomSerializer.Proxy<T
         } else {
             obj.message
         }
-        return ThrowableProxy(obj.javaClass.name, message, obj.stackTrace, obj.cause, obj.suppressed, extraProperties)
+        return if(shouldIncludeInternalInfo())
```
Comment (rick-r3) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/custom/ThrowableSerializer.kt]: I think I would conditionally figure out the stack trace, then have a single construction of the the `ThrowableProxy`

File: core/src/main/kotlin/net/corda/core/serialization/SerializationAPI.kt [hunk 3]
```diff
@@ -86,6 +86,10 @@ abstract class SerializationFactory {
     }
 }
 
+enum class CommonPropertyNames {
```
Comment (rick-r3) [inline core/src/main/kotlin/net/corda/core/serialization/SerializationAPI.kt]: We could keep this in `node-api` module, in an `internal` package.

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/custom/ThrowableSerializer.kt (hunk 1 above)
Comment (vkolomeyko) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/custom/ThrowableSerializer.kt]: sure, will change

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/custom/ThrowableSerializer.kt (hunk 2 above)
Comment (vkolomeyko) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/custom/ThrowableSerializer.kt]: will change

File: core/src/main/kotlin/net/corda/core/serialization/SerializationAPI.kt (hunk 3 above)
Comment (vkolomeyko) [reply core/src/main/kotlin/net/corda/core/serialization/SerializationAPI.kt]: Will change

**Pull request merged**